lib/misc.ml

```diff
@@ -165,79 +122,10 @@ let rec list_eq eq xs ys = match xs,ys with
   | ([],_::_)|(_::_,[]) -> false
   | x::xs,y::ys -> eq x y && list_eq eq xs ys
 
-(* Avoid String.lowercase warning *)
-let char_lowercase = function
-  | 'A' -> 'a'
-  | 'B' -> 'b'
-  | 'C' -> 'c'
-  | 'D' -> 'd'
-  | 'E' -> 'e'
-  | 'F' -> 'f'
-  | 'G' -> 'g'
-  | 'H' -> 'h'
-  | 'I' -> 'i'
-  | 'J' -> 'j'
-  | 'K' -> 'k'
-  | 'L' -> 'l'
-  | 'M' -> 'm'
-  | 'N' -> 'n'
-  | 'O' -> 'o'
-  | 'P' -> 'p'
-  | 'Q' -> 'q'
-  | 'R' -> 'r'
-  | 'S' -> 's'
-  | 'T' -> 't'
-  | 'U' -> 'u'
-  | 'V' -> 'v'
-  | 'W' -> 'w'
-  | 'X' -> 'x'
-  | 'Y' -> 'y'
-  | 'Z' -> 'z'
-  | c -> c
-
-let char_uppercase = function
-  | 'a' -> 'A'
-  | 'b' -> 'B'
-  | 'c' -> 'C'
-  | 'd' -> 'D'
-  | 'e' -> 'E'
-  | 'f' -> 'F'
-  | 'g' -> 'G'
-  | 'h' -> 'H'
-  | 'i' -> 'I'
-  | 'j' -> 'J'
-  | 'k' -> 'K'
-  | 'l' -> 'L'
-  | 'm' -> 'M'
-  | 'n' -> 'N'
-  | 'o' -> 'O'
-  | 'p' -> 'P'
-  | 'q' -> 'Q'
-  | 'r' -> 'R'
-  | 's' -> 'S'
-  | 't' -> 'T'
-  | 'u' -> 'U'
-  | 'v' -> 'V'
-  | 'w' -> 'W'
-  | 'x' -> 'X'
-  | 'y' -> 'Y'
-  | 'z' -> 'Z'
-  | c -> c
-
-let xcase c s =
-   let r =
-    Bytes.init (String.length s)
-      (fun k -> c (String.unsafe_get s k)) in
-   Bytes.unsafe_to_string r
-
-let lowercase s = xcase char_lowercase s
-let uppercase s = xcase char_uppercase s
-
-let capitalize s = match s with
-| "" -> assert false
-| _ ->
-    String.make 1 (char_uppercase s.[0]) ^
-    String.sub s 1 (String.length s-1)
+let char_uppercase = Char.uppercase_ascii
+let lowercase = String.lowercase_ascii
+let uppercase = String.uppercase_ascii
+let capitalize = String.capitalize_ascii
```

Comment:
Slightly different behaviour in the case of
```
Misc.capitalize ""
```
used to assert now it returns ""
Again the new behaviour makes sense to me, just checking.